sklearn/impute.py

```diff
@@ -643,17 +646,25 @@ def _impute_one_feature(self,
                                missing_row_mask)
         if self.sample_posterior:
             mus, sigmas = predictor.predict(X_test, return_std=True)
-            good_sigmas = sigmas > 0
             imputed_values = np.zeros(mus.shape, dtype=X_filled.dtype)
-            imputed_values[~good_sigmas] = mus[~good_sigmas]
-            mus = mus[good_sigmas]
-            sigmas = sigmas[good_sigmas]
+            # two types of problems: (1) non-positive sigmas, (2) mus outside
```

Comment:
@jnothman @benlawson This is my proposed solution to the problem of having `mus` outside of the legal `min_value` and `max_value`. However, after implementing this, now this test fails: `test_iterative_imputer_truncated_normal_posterior`. So something is off here...

Reply:
OK, what's happening is that the first prediction via `mus, sigmas = predictor.predict(X_test, return_std=True)` is outside of the min/max so it never gets to the truncated draw, and thus the imputations from `imputations = np.array([imputer.transform(X)[0][0] for _ in range(1000)])` are all 0.5.

Reply:
I changed the random seed, which allows the first draw to be between min/max bounds, and thus results in the test passing, I also turned down 1000 draws to 100 as it seemed unnecessarily slow.